t/config.t

```diff
@@ -88,6 +88,38 @@ subtest 'Everything but NoWarnings' => sub {
         is $config->ZONEMASTER_age_reuse_previous_test,                  600, 'default: ZONEMASTER.age_reuse_previous_test';
     };
 
+    lives_and {
+        my $text = q{
+            [DB]
+            engine = MySQL
+
+            [MYSQL]
+            host     = ::1
+            user     = mysql_user
+            password = mysql_password
+            database = mysql_database
+        };
+        my $config = Zonemaster::Backend::Config->parse( $text );
+        is $config->MYSQL_host, '::1', 'set: MYSQL.host';
+        is $config->MYSQL_port, 3306,  'default: MYSQL.port';
```

Comment:
Did you consider extending the first two `lives_and` clauses instead of creating new ones?

Reply:
Not at all. I made this choice because this is specific to different database engine, but I'm fine with checking all the default values in one place.

Reply:
I think it's simple and scalable to have all the defaults grouped together. I had a couple of design ideas when I wrote config.t but I guess I failed to communicate them.

When I had another look at config.t I realized a couple of other mistakes I made. I'll take another stab to improve it a little after you've merged this PR.